doc/man3/EVP_PKEY_CTX_ctrl.pod

```diff
@@ -67,8 +81,23 @@ B<pkeyutl>, B<genpkey> and B<req> commands.
 All the remaining "functions" are implemented as macros.
 
 The EVP_PKEY_CTX_set_signature_md() macro sets the message digest type used
-in a signature. It can be used with any public key algorithm supporting
-signature operations.
+in a signature. It can be used in the RSA, DSA and ECDSA algorithms.
+
+The EVP_PKEY_CTX_get_signature_md() macro gets the message digest type used in a
+signature. It can be used in the RSA, DSA and ECDSA algorithms.
+
+The EVP_PKEY_CTX_set_priv_key() macro sets the raw private key data to be used
+when during a key generation operation. Typically users would call
+L<EVP_PKEY_new_priv_key(3)> instead of this macro. It can be used with any of
```

Comment:
If you set the private (or public) key you're really not generating a (new) key, so that sentence doesn't make sense to me.

It's also not clear to me why we need this as a public function.

("when during" doesn't make sense.)

Reply:
I reworded this to make this clearer (and to try and explain what is meant by key generation better). Note: these aren't functions but macros. EVP_PKEY_CTX_set_mac_key() is already a public macro and EVP_PKEY_set_priv_key() is just a synonym for it (because it now covers more than just mac keys). The underlying ctrls *have* to be public. I could make the EVP_PKEY_set_priv_key() and EVP_PKEY_set_pub_key() macros private, but given that the ctrls are public it seems weird not too.